src/python/pants/init/engine_initializer.py

```diff
@@ -200,7 +200,10 @@ def run_console_rules(self, options_bootstrapper, goals, target_roots):
       use_colors=options_bootstrapper.bootstrap_options.for_global_scope().colors
     )
     for goal in goals:
-      goal_product = self.goal_map[goal]
+      try:
+        goal_product = self.goal_map[goal]
```

Comment:
This is a completely orthogonal change which should be in a separate PR, where singleton goals (in our monorepo, for example, we have `scalafix` registered as its own goal), and that's accidentally considered an "ambiguous goal" for the purposes of v2 console_rule resolution. This is relatively easy to repro, and is probably not hard to fix, but should definitely be in a separate change.